geopandas/tests/test_geocode.py

```diff
@@ -105,41 +105,47 @@ def test_prepare_result_none(self):
         row = df.loc['b']
         assert len(row['geometry'].coords) == 0
         assert np.isnan(row['address'])
-    
+
     def test_bad_provider_forward(self):
-        with pytest.raises(ValueError):
+        from geopy.exc import GeocoderNotFound
+        with pytest.raises(GeocoderNotFound):
             geocode(['cambridge, ma'], 'badprovider')
 
     def test_bad_provider_reverse(self):
-        with pytest.raises(ValueError):
+        from geopy.exc import GeocoderNotFound
+        with pytest.raises(GeocoderNotFound):
             reverse_geocode(['cambridge, ma'], 'badprovider')
 
     def test_forward(self):
-        with mock.patch('geopy.geocoders.googlev3.GoogleV3.geocode',
-                        ForwardMock()) as m:
-            g = geocode(self.locations, provider='googlev3', timeout=2)
-            assert len(self.locations) == m.call_count
-
-        n = len(self.locations)
-        assert isinstance(g, GeoDataFrame)
-        expected = GeoSeries(
-            [Point(float(x) + 0.5, float(x)) for x in range(n)],
-            crs=from_epsg(4326))
-        assert_geoseries_equal(expected, g['geometry'])
-        assert_series_equal(g['address'],
-                            pd.Series(self.locations, name='address'))
+        from geopy.geocoders import GoogleV3
+        for provider in ['googlev3', GoogleV3]:
```

Comment:
I was originally intending to use `pytest.mark.parametrize` here, but I haven't ever used that in a situation where protecting against a soft dependency (and thus not being able to simply mark it to use a Geocoder instance already imported from geopy here). While ideally the tests of the access via string or Geocoder argument would be separate, it struck me in this case as not being worth splitting these two tests into test support functions that are then called by separate test functions that specify the provider.

Reply:
yes, this is fine 
(I think in principle you can do a combination of different decorators (one that skips if geopy is not available, and then the parametrize), but the current PR keeps it more consistent with the rest of the test class)

Reply:
The wrinkle is in defining the `GoogleV3` instance to pass to `pytest.mark.parametrize`. That definition has to happen outside the test.